pkg/container/versioned/value_test.go

```diff
@@ -0,0 +1,289 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package versioned
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVersionedValue(t *testing.T) {
+	var value1 Value[[]uint32]
+
+	cv := NewVersionManager(func(keepVersion Version) {
+		Cleaner(&value1, keepVersion)
+	})
+
+	// Initially empty
+	handle, closer := cv.GetHandle("empty")
+	assert.Empty(t, GetValue(&value1, handle))
+	assert.True(t, closer())
+	// 2ns call does nothing
+	assert.False(t, closer())
+
+	// Add first value
+	handle1, closer1 := cv.GetHandle("first")
+	assert.Empty(t, GetValue(&value1, handle1))
+
+	current, next := cv.GetVersion()
+	assert.Less(t, current, next)
+
+	SetValueAtVersion(&value1, []uint32{100, 200}, next)
+	assert.True(t, cv.PublishVersion(current, next))
+
+	current, _ = cv.GetVersion()
+	assert.Equal(t, next, current)
+
+	// New value is invisible for the old handle
+	assert.Empty(t, GetValue(&value1, handle))
+
+	// But is visible for any new handles
+	handle2, closer2 := cv.GetHandle("second")
+
+	v := GetValue(&value1, handle2)
+	assert.Len(t, v, 2)
+	assert.Equal(t, []uint32{100, 200}, v)
+
+	// Set a new value
+	current, next = cv.GetVersion()
+	SetValueAtVersion(&value1, []uint32{100, 150, 200}, next)
+	assert.True(t, cv.PublishVersion(current, next))
+
+	// new handle sees the new value
+	handle3, closer3 := cv.GetHandle("third")
+	v = GetValue(&value1, handle3)
+	assert.Len(t, v, 3)
```

Comment:
The length assertions are redundant if equality is checked afterwards, and the error produced is probably better in the equal check?

Reply:
yes.